src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosUnbounding = {|
+  validatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosResources = {|
+  delegations: CosmosDelegation[],
+  redelegations: CosmosRedelegation[],
+  unbounding: CosmosUnbounding[],
+  delegatedBalance: BigNumber,
+  pendingRewardsBalance: BigNumber, 
+  unboundingBalance: BigNumber,
+  withdrawAddress: string,
+|};
+
+export type CosmosDelegationRaw = {|
+  validatorAddress: string,
+  amount: string,
+  pendingRewards: string,
+  isActive: boolean,
+|};
+
+export type CosmosRedelegationRaw = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosUnboundingRaw = {|
+  validatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosResourcesRaw = {|
+  delegations: CosmosDelegationRaw[],
+  redelegations: CosmosRedelegationRaw[],
+  unbounding: CosmosUnboundingRaw[],
+  delegatedBalance: string,
+  pendingRewardsBalance: string, 
+  unboundingBalance: string,
+  withdrawAddress: string,
+|};
+
+// NB this must be serializable (no Date, no BigNumber)
+export type CosmosValidatorItem = {|
+  validatorAddress: string,
+  name: string,
+  votingPower: number, // value from 0.0 to 1.0 (normalized percentage)
+  commission: number, // value from 0.0 to 1.0 (normalized percentage)
+|};
+
+// by convention preload would return a Promise of CosmosPreloadData
+export type CosmosPreloadData = {
+  validators: CosmosValidatorItem[]
```

Comment:
We can add a `RewardsState` object or something to hold the data necessary for reward rate computation.

We need :
Network parameters : 
- target bonded ratio
- community pool commission
- assumed time per block
- inflation rate change
- inflation max rate
- inflation min rate

Network dynamic values : 
- actual bonded ratio
- average time per block
- total supply of ATOM
- average daily fees
- current value of inflation on the network

(The only other value is the validator commission that we have in the validators array).

We'll just mock all these data to start, and I'll help with the js implementation later

Reply:
ok. @henri-ly could you help me adding the necessary parts?

Reply:
I will come with something this afternoon